File: Fibonacci/fibonacci.js

// one 
//Recursive

function Fibonacci(n){
    if(n <= 1){
        return n;
    }else{

        return Fibonacci (n - 1) + Fibonacci (n - 2);
    }
}
console.log(Fibonacci(10)); //output = 55



// tow

function Fibonacci(length) {
    let a = 0,
      b = 1,
      sum = 0
    for (let i = 2; i <= length; i++) {
      sum = a + b
      a = b
      b = sum
    }
    return sum
  }

File: Fibonacci/fibonacci.py

# one 
# Recursive

def Fibonacci (n):
    if n <= 1:
        return n 
    else:
        return Fibonacci(n - 1) + Fibonacci (n - 2)
    
print(Fibonacci(10))  #output = 55     


# tow
# iterative
def fibonacci(n):
    a, b = 0, 1
    for _ in range(n):
        yield a
        a, b = b, a + b

n = 10 
fib_sequence = list(fibonacci(n))

print(fib_sequence)    #[0, 1, 1, 2, 3, 5, 8, 13, 21, 34].
